src/AElf.CSharp.CodeOps/Patchers/Module/ResetFieldsMethodInjector.cs

```diff
@@ -1,42 +1,49 @@
 using System.Collections.Generic;
+using System.IO;
 using System.Linq;
-using Google.Protobuf.Reflection;
+using AElf.Kernel.CodeCheck;
 using Mono.Cecil;
 using Mono.Cecil.Cil;
+using Mono.Cecil.Rocks;
+using MethodAttributes = Mono.Cecil.MethodAttributes;
 
 namespace AElf.CSharp.CodeOps.Patchers.Module;
 
-public class ResetFieldsMethodInjector : IPatcher<ModuleDefinition>
+public partial class ResetFieldsMethodInjector : IPatcher<ModuleDefinition>
 {
     public bool SystemContactIgnored => false;
-        
+
     public void Patch(ModuleDefinition module)
     {
+        var checker = new ExecutionObserverProxyChecker(module);
+        var disallowedFields = module.GetAllTypes().SelectMany(t => t.Fields).Where(f => f.IsDisallowedStaticField())
+            .Where(f => !checker.IsObserverFieldThatRequiresResetting(f)).ToList();
+        if (!disallowedFields.IsNullOrEmpty())
+        {
+            throw new InvalidCodeException("Static field is not allowed in user code.");
+        }
```

Comment:
Should there be CodeCheck logic in the CodePatch phase?

Reply:
Opened https://github.com/AElfProject/AElf/issues/3389 to track it properly.